tls/s2n_client_hello.c

```diff
@@ -194,6 +194,7 @@ static int s2n_parse_client_hello(struct s2n_connection *conn)
 
     /* This is going to be our default if the client has no preference. */
     conn->secure.server_ecc_evp_params.negotiated_curve = ecc_pref->ecc_curves[0];
+    conn->secure.server_ecc_evp_params.evp_pkey = NULL;
```

Comment:
Does this introduce a potential memory leak? Don't you need to free the key if it exists?

And why would it exist at this point?

Reply:
At this point ` conn->secure.server_ecc_evp_params.evp_pkey` MUST be null as nothing is initialized prior to the client_hello_recv.

Reply:
If it MUST be null, we need to error on it being not-null. If you just silently set it to null, that introduces a potential memory leak.